IPython/lib/pretty.py

```diff
@@ -500,6 +516,75 @@ def remove(self, group):
             pass
 
 
+class RawText:
+    """ Object such that ``p.pretty(RawText(value))`` is the same as ``p.text(value)``.
+
+    An example usage of this would be to show a list as binary numbers, using
+    ``p.pretty([RawText(bin(i)) for i in integers])``.
+    """
+    def __init__(self, value):
+        self.value = value
+
+    def _repr_pretty_(self, p, cycle):
+        p.text(self.value)
+
+
+class CallExpression:
+    """ Object which emits a line-wrapped call expression in the form `__name(*args, **kwargs)` """
+    def __init__(__self, __name, *args, **kwargs):
+        # dunders are to avoid clashes with kwargs, as python's name manging
+        # will kick in.
+        self = __self
+        self.name = __name
+        self.args = args
+        self.kwargs = kwargs
+
+    @classmethod
+    def factory(cls, name):
+        def inner(*args, **kwargs):
+            return cls(name, *args, **kwargs)
+        return inner
```

Comment:
You need to be doing this only to curry when `name` right ? Would you consider always currying on `name` ? Like moving `self.args = args` and `self.kwargs = kwargs` into for example a `__call__` ?

Reply:
I don't really like the idea of the `__call__()` mutating the object, but if there's a functionally-pure way to require the currying (make the class name private?), I'm not opposed to that.

Reply:
One simple option would be:
```python
class _NamedCallExpression
    ... # as before

def CallExpression(name):
    # contents of `def factory`
```